src/AppInstallerCommonCore/Synchronization.cpp

```diff
@@ -259,4 +262,48 @@ namespace AppInstaller::Synchronization
 
         return result;
     }
+
+    CrossProcessInstallLock::CrossProcessInstallLock()
+    {
+        m_mutex.create(L"WinGetCrossProcessInstallLock", 0, SYNCHRONIZE);

```

Comment:
Should we be worried about some third party acquiring this named mutex and just making winget unusable?

Reply:
We could add a random suffix to the mutex (and any other named mutexes), but we would have to write that information into a location that the unelevated user token can access.  While it would make it slightly harder to execute this attack, it wouldn't be any harder from a security perspective.

I suppose that we could put a conditional ACE on it to require our package identity (when packaged) to even read from it.  Even that is just another minor bump in the road that can be overcome with more code (but not any more security access).